client/gutenberg/extensions/map/edit.js

```diff
@@ -0,0 +1,260 @@
+		noticeOperations.removeAllNotices();
+		noticeOperations.createErrorNotice( message );
+	};
+	render() {
+		const { className, setAttributes, attributes, noticeUI, notices } = this.props;
+		const { map_style, map_details, points, zoom, map_center, marker_color, align } = attributes;
+		const { addPointVisibility, api_key, apiKeyControl, apiState } = this.state;
+		const inspectorControls = (
+			<Fragment>
+				<BlockControls>
+					<BlockAlignmentToolbar
+						value={ align }
+						onChange={ this.updateAlignment }
+						controls={ [ 'center', 'wide', 'full' ] }
+					/>
+					<Toolbar>
+						<IconButton
+							icon={ settings.markerIcon }
+							label="Add a marker"
+							onClick={ () => this.setState( { addPointVisibility: true } ) }
+						/>
+					</Toolbar>
+				</BlockControls>
+				<InspectorControls>
+					<PanelBody title={ __( 'Map Theme', 'jetpack' ) }>
+						<MapThemePicker
+							value={ map_style }
+							onChange={ value => setAttributes( { map_style: value } ) }
+							options={ settings.map_styleOptions }
+						/>
+						<ToggleControl
+							label={ __( 'Show street names', 'jetpack' ) }
+							checked={ map_details }
+							onChange={ value => setAttributes( { map_details: value } ) }
+						/>
+					</PanelBody>
+					<PanelColorSettings
+						title={ __( 'Colors', 'jetpack' ) }
+						initialOpen={ true }
+						colorSettings={ [
+							{
+								value: marker_color,
+								onChange: value => setAttributes( { marker_color: value } ),
+								label: 'Marker Color',
+							},
+						] }
+					/>
+					{ points.length ? (
+						<PanelBody title={ __( 'Markers', 'jetpack' ) } initialOpen={ false }>
+							<Locations
+								points={ points }
+								onChange={ value => {
+									setAttributes( { points: value } );
+								} }
+							/>
+						</PanelBody>
+					) : null }
+					<PanelBody title={ __( 'Mapbox Access Token', 'jetpack' ) } initialOpen={ false }>
+						<TextControl
+							label={ __( 'Mapbox Access Token', 'jetpack' ) }
+							value={ apiKeyControl }
+							onChange={ value => this.setState( { apiKeyControl: value } ) }
+						/>
+						<ButtonGroup>
+							<Button type="button" onClick={ this.updateAPIKey } isDefault>
+								{ __( 'Update Key' ) }
+							</Button>
+							<Button type="button" onClick={ this.removeAPIKey } isDefault>
+								{ __( 'Remove Key' ) }
+							</Button>
+						</ButtonGroup>
+					</PanelBody>
+				</InspectorControls>
+			</Fragment>
+		);
+		const placholderAPIStateLoading = (
+			<Placeholder icon={ settings.icon }>
+				<Spinner />
+			</Placeholder>
+		);
+		const getAPIInstructions = sprintf(
+			"<p>Before you use a map block, you will need to get a key from <a href='%1$s'>Mapbox</a>. You will only have to do this once.</p><p>Go to <a href='%1$s'>Mapbox</a> and either create an account or sign in. Once you sign in, locate and copy the default access token. Finally, paste it into the token field below.</p>",
```

Comment:
needs translation?

Reply:
Sadly HTML/link within translated text doesn't work yet in Gutenberg. :-(

https://github.com/WordPress/gutenberg/issues/9846

E.g. for Simple Payments block I had to place links separately which was kinda annoying:

```jsx
<p>
  { __( 'Lorem ipsum.' ) }
  <a href="#">{ __( 'Clickety click!' ) }</a>
</p>
```

Reply:
Going to deal with some of the i18n stuff in a future PR, after we've determined which service we're using.